SU2_CFD/src/fluid/CUserDefinedTCLib.cpp

```diff
@@ -0,0 +1,1667 @@
+    //NO
+    Omega11(2,0,0) = -6.8237776E-03;  Omega11(2,0,1) = 1.4360616E-01;   Omega11(2,0,2) = -1.1922240E+00;  Omega11(2,0,3) = 1.2433086E+03;
+    Omega11(2,1,0) = -6.8508672E-03;  Omega11(2,1,1) = 1.5524564E-01;   Omega11(2,1,2) = -1.3479583E+00;  Omega11(2,1,3) = 2.0037890E+03;
+    Omega11(2,2,0) = -7.4942466E-03;  Omega11(2,2,1) = 1.6626193E-01;   Omega11(2,2,2) = -1.4107027E+00;  Omega11(2,2,3) = 2.3097604E+03;
+    Omega11(2,3,0) = -1.4719259E-03;  Omega11(2,3,1) = 1.8446968E-02;   Omega11(2,3,2) = -2.6460411E-01;  Omega11(2,3,3) = 1.0911124E+02;
+    Omega11(2,4,0) = -1.0066279E-03;  Omega11(2,4,1) = 1.1029264E-02;   Omega11(2,4,2) = -2.0671266E-01;  Omega11(2,4,3) = 8.2644384E+01;
+    //N
+    Omega11(3,0,0) = -8.3493693E-03;  Omega11(3,0,1) = 1.7808911E-01;   Omega11(3,0,2) = -1.4466155E+00;  Omega11(3,0,3) = 1.9324210E+03;
+    Omega11(3,1,0) = -1.0608832E-03;  Omega11(3,1,1) = 1.1782595E-02;   Omega11(3,1,2) = -2.1246301E-01;  Omega11(3,1,3) = 8.4561598E+01;
+    Omega11(3,2,0) = -1.4719259E-03;  Omega11(3,2,1) = 1.8446968E-02;   Omega11(3,2,2) = -2.6460411E-01;  Omega11(3,2,3) = 1.0911124E+02;
+    Omega11(3,3,0) = -7.7439615E-03;  Omega11(3,3,1) = 1.7129007E-01;   Omega11(3,3,2) = -1.4809088E+00;  Omega11(3,3,3) = 2.1284951E+03;
+    Omega11(3,4,0) = -5.0478143E-03;  Omega11(3,4,1) = 1.0236186E-01;   Omega11(3,4,2) = -9.0058935E-01;  Omega11(3,4,3) = 4.4472565E+02;
+    //O
+    Omega11(4,0,0) = -8.3110691E-03;  Omega11(4,0,1) = 1.9617877E-01;   Omega11(4,0,2) = -1.7205427E+00;  Omega11(4,0,3) = 4.0812829E+03;
+    Omega11(4,1,0) = -3.7969686E-03;  Omega11(4,1,1) = 7.6789981E-02;   Omega11(4,1,2) = -7.3056809E-01;  Omega11(4,1,3) = 3.3958171E+02;
+    Omega11(4,2,0) = -1.0066279E-03;  Omega11(4,2,1) = 1.1029264E-02;   Omega11(4,2,2) = -2.0671266E-01;  Omega11(4,2,3) = 8.2644384E+01;
+    Omega11(4,3,0) = -5.0478143E-03;  Omega11(4,3,1) = 1.0236186E-01;   Omega11(4,3,2) = -9.0058935E-01;  Omega11(4,3,3) = 4.4472565E+02;
+    Omega11(4,4,0) = -4.2451096E-03;  Omega11(4,4,1) = 9.6820337E-02;   Omega11(4,4,2) = -9.9770795E-01;  Omega11(4,4,3) = 8.3320644E+02;
+    break;
+  }
+
+  if (ionization) { nHeavy = nSpecies-1; nEl = 1; }
+  else            { nHeavy = nSpecies;   nEl = 0; }
+}
+
+CUserDefinedTCLib::~CUserDefinedTCLib(){}
+  
+void CUserDefinedTCLib::SetTDStateRhosTTv(vector<su2double> val_rhos, su2double val_temperature, su2double val_temperature_ve){
+
+  rhos = val_rhos;
+  T    = val_temperature;
+  Tve  = val_temperature_ve;
+
+  Density = 0.0;
+  for (iSpecies = 0; iSpecies < nSpecies; iSpecies++)
+    Density += rhos[iSpecies];
+
+  Pressure = GetPressure();
+
+}
+
+vector<su2double> CUserDefinedTCLib::GetSpeciesCvTraRot(){
+ 
+  for (iSpecies = 0; iSpecies < nHeavy; iSpecies++)
+    Cvtrs[iSpecies] = (3.0/2.0 + RotationModes[iSpecies]/2.0) * Ru/MolarMass[iSpecies];
+
+  return Cvtrs;
+}
```

Comment:
There might be a good reason for this, but:
When returning vectors or other expensive-to-copy types, by value as done here, it is usually better to declare the variable you want to return locally, that way the copy is avoided, instead the result is "moved" into the calling function.

If the objective is to allocate space for these vectors only once in the fluid model, then they should be returned by `const&`.